qiskit_dynamics/type_utils.py

```diff
@@ -353,7 +355,7 @@ def isinstance_qutip_qobj(obj):
 
 
 # pylint: disable=too-many-return-statements
-def to_array(op: Union[Operator, Array, List[Operator], List[Array], spmatrix], no_iter=False):
```

Comment:
The `spmatrix` class is abstract one. 
- leave it to the type hint.
- delete it and we need to register additional concrete class such as `csc_matrix` to ArrayLike.